pkg/vmprovider/providers/vsphere/virtualmachine/backup.go

```diff
@@ -72,10 +92,41 @@ func getEncodedVMBootstrapData(bootstrapData map[string]string) (string, error)
 		return "", nil
 	}
 
-	bootstrapDataYaml, err := yaml.Marshal(bootstrapData)
+	bootstrapDataJSON, err := json.Marshal(bootstrapData)
+	if err != nil {
+		return "", err
+	}
+
+	return util.EncodeGzipBase64(string(bootstrapDataJSON))
+}
+
+func getEncodedVMDiskData(
+	ctx goctx.Context, vcVM *object.VirtualMachine) (string, error) {
+	resVM := res.NewVMFromObject(vcVM)
+	disks, err := resVM.GetVirtualDisks(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	diskData := []VMDiskData{}
+	for _, device := range disks {
+		if disk, ok := device.(*types.VirtualDisk); ok {
+			vmDiskData := VMDiskData{}
+			if disk.VDiskId != nil {
+				vmDiskData.VDiskID = disk.VDiskId.Id
+			}
+			if b, ok := disk.Backing.(*types.VirtualDiskFlatVer2BackingInfo); ok {
+				vmDiskData.FileName = b.FileName
+			}
+
+			diskData = append(diskData, vmDiskData)
```

Comment:
Does this need to be added if both fields are empty in the `vmDiskData`?

Reply:
Updated to skip empty `vmDiskData` element.